rclcpp/src/rclcpp/executors/multi_threaded_executor.cpp

```diff
@@ -42,15 +42,17 @@ MultiThreadedExecutor::spin()
   }
   RCLCPP_SCOPE_EXIT(this->spinning.store(false); );
   std::vector<std::thread> threads;
-  {
+  size_t thread_id = 1;
+  if (number_of_threads_ > 0) {
     std::lock_guard<std::mutex> wait_lock(wait_mutex_);
-    size_t thread_id = 1;
-    for (size_t i = number_of_threads_; i > 0; --i) {
+    for (size_t i = number_of_threads_ - 1; i > 0; --i) {
```

Comment:
Since the abort condition is `i > 0` it looks like the loop only runs `N - 2` times. Namely (N-1, N-2, ..., 2, 1)?

What about simply counting upwards. You could even use the same variable for `thread_id` and `i`.